pkg/apis/controller/v1beta1/appwrapper.go

```diff
@@ -206,6 +206,10 @@ type AppWrapperStatus struct {
 	// +optional
 	Pending int32 `json:"pending,omitempty" protobuf:"bytes,1,opt,name=pending"`
 
+	// The number of pending pods that failed scheduling
+	// +optional
+	PendingPodsFailedSchd int32 `json:"pending,omitempty" protobuf:"bytes,1,opt,name=pending"`
```

Comment:
In order to support providing a reason for preemption changing this field to a struct tracking the conditions instead of an int will be helpful.  The new field `PendingDueToScheduler[] FailedSchedulerCondition` and a struct `FailedSchedulerCondition` can be { PodName string, v1.PodCondition }

Reply:
Created a new field called:

```	
// Represents the latest available observations of pods under appwrapper
PendingPodConditions map[string][]v1.PodCondition `json:"pendingpodconditions,omitempty"

```
For `type AppWrapperStatus struct`

Would you agree?

Reply:
resolving, thanks